omegaconf/base.py

```diff
@@ -362,19 +362,23 @@ def resolve_simple_interpolation(
         self,
         key: Any,
         inter_type: str,
-        inter_key: str,
+        inter_key: Tuple[Any, ...],
```

Comment:
Can you explain why inter_key can now be a Tuple of anything?

Reply:
This is for resolver interpolations (`${f:x,y}`). Before, we would pass `inter_key="x,y"` and this string would be split into the `["x", "y"]` list within this function. However, now we want to be able to pass non-string arguments, so `inter_key` would be the tuple `(x, y)`.

Probably that `inter_key` is not a very good name for this argument. I assume it was initially called this way for node interpolations (`${x.y}`) where it refers to a key, but for resolver interpolations it can be confusing. I could split it in two mutually exclusive arguments instead to make it clearer, or into two different functions. Do you have any preference?

Reply:
Yeah, this is probably an existing confusion there.
See if you can clear it up. can you describe how you would split the function and what would be the responsibility of each of the two functions?

Reply:
I would have two different functions `_resolve_node_interpolation()` and `_resolve_resolver_interpolation()` (the second name is a bit unfortunate, but it's actually the one that makes more sense IMO to be consistent with the terminology used elsewhwere)
The first one would essentially contain the code below `if inter_type is None:` and the second one the code below `else:`